File: android/entities/src/main/java/com/tiagohs/entities/main_topics/MainTopicItem.kt
package com.tiagohs.entities.main_topics

import com.google.gson.annotations.SerializedName
import com.tiagohs.entities.image.Image
import com.tiagohs.entities.Quote
import com.tiagohs.entities.Sumario
import com.tiagohs.entities.enums.ViewPosition
import java.io.Serializable

data class MainTopicItem(
    @SerializedName("id")
    val id: Int,

    @SerializedName("title")
    val title: String,

    @SerializedName("subtitle")
    val subtitle: String,

    @SerializedName("description")
    val description: String,

    @SerializedName("image")
    val image: Image,

    @SerializedName("blocked")
    val blocked: Boolean,

    @SerializedName("presentation_image")
    val presentationImage: Image? = null,

    @SerializedName("title_color")
    val titleColor: String? = null,

    @SerializedName("title_background_color")
    val titleBackgroundColor: String? = null,

    @SerializedName("quote_position")
    val quotePosition: ViewPosition,

    @SerializedName("quote")
    val quote: Quote,

    @SerializedName("sumario_list")
    var sumarioList: List<Sumario>

): MainTopic(), Serializable {

}

File: ios/HistoryCinema/HistoryCinema/Entities/LocalFiles/MainTopics/MainTopicItem.swift
//
//  MainTopicItem.swift
//  HistoryCinema
//
//  Created by Tiago Henrique da Silva on 27/02/21.
//

import Foundation
import ObjectMapper

class MainTopicItem: MainTopic {
    var id: Int!
    var title: String!
    var subtitle: String!
    var description: String!
    var image: LocalImage!
    var blocked: Bool!
    var isNew: Bool!
    var presentationImage: LocalImage?
    var titleColor: String?
    var titleBackgroundColor: String?
    var color: String?
    var quotePosition: ViewPosition!
    var quote: Quote!
    var sumarioList: [SummaryModel]?
    
    required init?(map: Map) {
        super.init(map: map)
    }

    override func mapping(map: Map) {
        super.mapping(map: map)
        
        quotePosition                               <- (map["quote_position"], EnumTransform<ViewPosition>())
        id                                          <- map["id"]
        title                                       <- map["title"]
        subtitle                                    <- map["subtitle"]
        description                                 <- map["description"]
        image                                       <- map["image"]
        blocked                                     <- map["blocked"]
        isNew                                       <- map["is_new"]
        presentationImage                           <- map["presentation_image"]
        titleColor                                  <- map["title_color"]
        titleBackgroundColor                        <- map["title_background_color"]
        color                                       <- map["color"]
        quotePosition                               <- map["quote_position"]
        quote                                       <- map["quote"]
    }
}
